### PR TITLE
docs: fill out static folder stub

### LINKED ISSUE #9222: Content migration - fill out "Static Folder" stub
## Issue Context

This issue is part of the content migration project (#8103), which is meant to take existing content and *repurpose* it so it's easier to find in the docs.

### Motivation

We are doing this because many people have written excellent content that we want to canonize and maintain so more people can find it and benefit from it.

### Review the Gatsby docs approach

- [ ] Read the Style Guide and follow it in order for your PR to be accepted. Pay special attention to the templates. 
  - Style Guide: https://www.gatsbyjs.org/docs/gatsby-style-guide/
  - Docs Templates: https://www.gatsbyjs.org/docs/templates/
  - Example of a similar article: https://www.gatsbyjs.org/docs/working-with-images/

### Draft the doc

- [ ] Add to the **Static Folder** [stub article](https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/static-folder.md)
- [ ] Use the following reference material as a primary starting point to draft the new doc:
  - [Using the static folder](https://www.gatsbyjs.org/docs/adding-images-fonts-files/#using-the-static-folder).
  - [When to use the static folder](https://www.gatsbyjs.org/docs/adding-images-fonts-files/#when-to-use-the-static-folder)
- [ ] Make sure you rewrite and adjust the content-- Add introductory language, etc. Make it a cohesive doc (not just copy/paste). Follow the format listed in these resources:
  - Docs Templates: https://www.gatsbyjs.org/docs/templates/
  - Example of a guide to imitate: https://www.gatsbyjs.org/docs/deploying-to-s3-cloudfront
- [ ] Add a sentence to the [Adding images fonts and files](https://www.gatsbyjs.org/docs/adding-images-fonts-files/) page, linking over to the new "static folder" doc page. (That way the content is linked, but not duplicated)
- [ ] Find the stub in the [docs sidebar](https://github.com/gatsbyjs/gatsby/blob/master/www/src/data/sidebars/doc-links.yaml), and remove the asterisk next to the title, (which will indicate that the doc is no longer a stub).

### Open a pull request
- [ ] Open a pull request with your work, including the words "closes #9222" in the pull request description.

### PR DESCRIPTION
**Pull request by rosariopfernandes**
closes #9222

### CODE REVIEW

File: docs/docs/static-folder.md [hunk 1]
```diff
@@ -2,7 +2,73 @@
...
+[images, and fonts](/docs/adding-images-fonts-files/#adding-images-and-fonts) from JavaScript. The `static`
+folder is useful as a workaround for a number of less common cases:
+
+- You need a file with a specific name in the build output, such as
+  [`manifest.webmanifest`](https://developer.mozilla.org/en-US/docs/Web/Manifest).
+- You have thousands of images and need to dynamically reference their paths.
+- You want to include a small script like
+  [`pace.js`](http://github.hubspot.com/pace/docs/welcome/) outside of the
+  bundled code.
+- Some library may be incompatible with Webpack and you have no other option but
```
Comment (shannonbux) [inline docs/docs/static-folder.md]: libraries?

**Comment by shannonbux** [comment]
@rosariopfernandes looks great! Thanks!

**Comment by LekoArts** [comment]
Duplicate: https://github.com/gatsbyjs/gatsby/pull/7038

**Comment by amberleyromo** [comment]
@rosariopfernandes looks like -- sorry about that! It was included on @shannonbux's content migration issue, and I didn't double check.

**Pull request merged**